pyproject.toml

```diff
@@ -0,0 +1,172 @@
+    "doctr",
+    "flake8",
+    "ipython",
+    "ipywidgets",
+    "jinja2",
+    "lmfit",
+    "matplotlib >=3.5.0",
+    "mongoquery",
+    "mypy",
+    "myst-parser",
+    "networkx",
+    "nose",
+    "numpydoc",
+    "ophyd",
+    "pandas",
+    "pims",
+    "pipdeptree",
+    "pre-commit",
+    "pydata-sphinx-theme>=0.12",
+    "pyepics",
+    "pyqt5",
+    "pytest",
+    "pytest-cov",
+    "pytest-faulthandler",
+    "pyyaml",
+    "pyzmq",
+    "requests",
+    "ruff",
+    "scikit-image",
+    "scipy",
+    "sphinx-autobuild",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "sphinx_rtd_theme",
+    "streamz",
+    # These suitcases are test deps of databroker which we need to access
+    # databroker fixtures.
+    "suitcase-jsonl",
+    "suitcase-mongo",
+    "suitcase-msgpack",
+    "tifffile",
+    "tox-direct",
+    "types-mock",
+]
+ipython = ["ipython"]
+zmq = ["pyzmq"]
+common = ["ophyd", "databroker"]
+tools = ["doct", "lmfit", "tifffile", "historydict"]
+streamz = ["streamz"]
+plotting = ["matplotlib"]
+cmd = ["colorama"]
+olog = ["jinja2"]
+all = ["bluesky[dev,ipython,zmq,common,tools,streamz,plotting,cmd,olog]"]
+
+[project.scripts]
+bluesky = "bluesky.__main__:main"
+
+[project.urls]
+GitHub = "https://github.com/bluesky/bluesky"
+
+[[project.authors]] # Further authors may be added by duplicating this section
+name = "danielballan"
+
+
+[tool.setuptools_scm]
+write_to = "src/bluesky/_version.py"
+
+[tool.mypy]
+ignore_missing_imports = true # Ignore missing stubs in imported modules
+
+[tool.pytest.ini_options]
+# Run pytest with all our checkers, and don't spam us with massive tracebacks on error
+addopts = """
+    --tb=native -vv --doctest-modules --doctest-glob="*.rst"
+    """
+# https://iscinumpy.gitlab.io/post/bound-version-constraints/#watch-for-warnings
+filterwarnings = "error"
+# Doctest python code in docs, python code in src docstrings, test functions in tests
+testpaths = "docs src tests"
+
+[tool.coverage.run]
+data_file = "/tmp/bluesky.coverage"
+
+[tool.coverage.report]
+exclude_lines = ["if __name__ == '__main__':"]
+
+[tool.coverage.paths]
+# Tests are run from installed location, map back to the src directory
+source = ["src", "**/site-packages/"]
+
+# tox must currently be configured via an embedded ini string
+# See: https://github.com/tox-dev/tox/issues/999
+[tool.tox]
+legacy_tox_ini = """
+[tox]
+skipsdist=True
+
+[testenv:{pre-commit,type-checking,tests,docs}]
+# Don't create a virtualenv for the command, requires tox-direct plugin
+direct = True
+passenv = *
+allowlist_externals =
+    pytest
+    pre-commit
+    mypy
+    sphinx-build
+    sphinx-autobuild
+commands =
+    pre-commit: pre-commit run --all-files {posargs}
+    type-checking: mypy src tests {posargs}
+    tests: pytest --cov=bluesky --cov-report term --cov-report xml:cov.xml {posargs}
+    docs: sphinx-{posargs:build -EW --keep-going} -T docs build/html
+"""
+
+[tool.ruff]
+src = ["src", "tests"]
+line-length = 115
```

Comment:
Do we want 115 character lines?

Reply:
Can make a ticket to change it later, idea was to minimize number of ruff errors to something manageable for one PR